### ISSUE TITLE
Multifox 2 forgets the saved accounts

### ISSUE THREAD

**Issue by Drugoy** [hultmann/multifox#38]
I am using latest Nightly and MultiFox 2.0b4

**Comment by Drugoy** [comment]
p.s.: I'm using latest Fx Nightly (9.0)

**Comment by hultmann** [comment]
Works as expected in Firefox 6. I'll try with Firefox 9 when I resume Multifox development.

**Comment by hultmann** [comment]
Marking #39 as a duplicate.

**Comment by 0rt** [comment]
encounter same isssues now and then, trying to induce a reproducible processes.

**Comment by hultmann** [comment]
Is this still happening? Using Nightly 10, Gmail/GReader/Twitter/Hotmail keep me logged in after a restart.

**Comment by Drugoy** [comment]
Seems like it is working fine for me now. I will test more and will write here if I'll experience that bug again.
Do you plan to work on multifox? it has not been updated for long, I can't find it on AMO and your site throws error 500.

**Comment by hultmann** [comment]
Yes, I use Multifox all the time and my plan is to finish v2.0. Hopefully it will happen sometime in 2012.

I'm closing this bug. Feel free to reopen if necessary.

**Comment by Drugoy** [comment]
It doesn't log me out anymore. But instead it just forgets it's own accounts.

**Comment by Drugoy** [comment]
Where does your extension store it's data? I clean unprotected (not whitelisted) cookies periodically. And Better Privacy cleans flash cookies too.

**Comment by Drugoy** [comment]
Multifox forgets saved accounts.
Latest Nightly, MultiFox 2.0b4

> I'm closing this bug. Feel free to reopen if necessary.

I don't have enough rights to.

**Comment by hultmann** [comment]
Thanks, I'll get back into development mode in a few weeks.

**Comment by Drugoy** [comment]
Is there any progress?
I didn't notice any changes since october 2011... please, don't give up the development.

**Comment by hultmann** [comment]
Actually I'm working on Multifox right now. However, I need a way to reproduce this bug.

> Where does your extension store it's data?

Mixed with default cookies/localStorage.

**Comment by Drugoy** [comment]
Even a new version of your extension is still useless to me due to this bug :(

**Comment by Drugoy** [comment]
why do you store all sites' cookies in a sitename.com.asdassdadasasdasd-asdasdasdas-asdasdsdadas.multifox-auth-2 format instead of asddaasdasdasdasddas-asddasdasdas-asdasdadsasd.multifox-auth-2.sitename.com?

**Comment by Drugoy** [comment]
I added "multifox-auth-2" to the cookies exceptions as "allow", and now everything works perfectly fine.
Why didn't I take a look at the cookies earlier..? :(

**Comment by hultmann** [comment]
> I added "multifox-auth-2" to the cookies exceptions as "allow", and now everything works perfectly fine.
> Why didn't I take a look at the cookies earlier..? :(

I'm not sure what you are trying to do...

> why do you store all sites' cookies in a sitename.com.asdassdadasasdasd-asdasdasdas-asdasdsdadas.multifox-auth-2 format instead of asddaasdasdasdasddas-asddasdasdas-asdasdadsasd.multifox-auth-2.sitename.com?

Because Firefox limits the number of cookies per TLD (or host, don't remember). Using a scheme like multifox-auth-2.sitename.com would make this bug reappear.

**Comment by Drugoy** [comment]
I opened "Menu > Tools > Options > Privacy > History - Firefox Will: [use custom settings for history] > [exceptions]" and there added a new rule (exception), where "multifox-auth-2" was set as allowed (I.e. I allowed Firefox to store the cookies like "*.multifox-auth-2" which solved this issue for me.

> Because Firefox limits the number of cookies per TLD (or host, don't remember). Using a scheme like multifox-auth-2.sitename.com would make this bug reappear.

But in your case Firefox regards "multifox-auth-2" as TLD and thus - it should also apply some limits to the number of cookies for it. Then why doesn't it? Could you provide a bug number in bugzilla to the corresponding bug?

**Comment by hultmann** [comment]
Why are all these cookies blocked? I would like to reproduce this bug...

> But in your case Firefox regards "multifox-auth-2" as TLD and thus - it should also apply some limits to the number of cookies for it. Then why doesn't it? Could you provide a bug number in bugzilla to the corresponding bug?

From my tests, aaa.xxx and bbb.xxx don't share the limits for cookies. Don't know if there's any bug related to that.

Hopefully these workarounds will go away with https://bugzilla.mozilla.org/show_bug.cgi?id=756648

**Comment by Drugoy** [comment]
> Why are all these cookies blocked? I would like to reproduce this bug...

I'm browsing in the paranoid mode: I wanted to block all the cookies, except whitelisted ones.
But since some stupid sites actually REQUIRE you to have cookies allowed to just read their pages - I had to do so,  but I also have installed an add-on that just deletes non-whitelisted cookies once in half an hour.

So to fix this issue I just had to whitelist "multifox-auth-2".
It was stupid of me to not understand earlier the reason of the issue I experienced.

I have a few quite general question about how Multifox works, would you mind to shed some light for me on that?
What if a user has 3 tabs: two with facebook (logged into two different accounts) and one with some news site, where you are allowed to leave a comment to the news, by logging to facebook (instead of creating account on that news site). And user decides to leave a comment there.
Will the news site hook some facebook account (that you are logged into) or not? <i>\- my guess is that it won't, since the user will have no facebook.com cookies, because they were spoofed by multifox into facebook.com.xxx.multifox-auth-2 and facebook.com.yyy.multifox-auth-2. But that is just my guess, I didn't try. But then why does "IGNOREME" appears, which you've said indicates that this page is using 3rd party site cookies that you have a multifox account for?</i>
If it will - in that case, which one will get used and why not the other?

**Comment by hultmann** [comment]
Currently the news site will pick up a logged in Facebook account (which one? it's almost random).

In the next weeks, Multifox will be aware about 3rd-party default accounts (via IGNOREME button). e.g. Facebook/Twitter/Google widgets can always use an anonymous account. If you open Gmail it will use a logged in account.

**Comment by Drugoy** [comment]
> So to fix this issue I just had to whitelist "multifox-auth-2".
> It was stupid of me to not understand earlier the reason of the issue I experienced.

I forgot to later mention that it didn't help: multifox keeps forgetting accounts.

**Comment by Drugoy** [comment]
I'm getting randomly logged off from the sites and/or Multifox also forgets logins to them.

**Comment by Drugoy** [comment]
@hultmann, since bug 756648 now has "resolved fixed" status - could you, please, change the cookies urls to the format I once suggested (multifox-auth-2.account-related-data.sitename.com instead of sitename.com.account-related-data.multifox-auth-2)?

**Comment by Drugoy** [comment]
the problem is that if you add "multifox-auth-2" as "allow" to your cookies exceptions list - they cookies will still get deleted if you chose to delete cookies at the end of the session.
Probably, Firefox treats this rule as wrong one, since it probably recognizes it as just a TLD, and probably one can't just add a whole TLD to the exceptions list.

**Comment by hultmann** [comment]
about cookie name: I still don't know how to use jars from bug 756648. it seems bug 811385 uses it, I'm not sure if it would be applicable to web pages.

PS: I believe cookies from jars are ignored by exceptions list (and invisible to cookie viewer).
